ambry-frontend/src/main/java/com.github.ambry.frontend/AmbrySecurityService.java

```diff
@@ -118,19 +118,19 @@ public AmbrySecurityService(FrontendConfig frontendConfig, FrontendMetrics front
             responseChannel.setStatus(ResponseStatus.Ok);
             RestUtils.SubResource subResource = RestUtils.getBlobSubResource(restRequest);
             if (subResource == null) {
+              responseChannel.setHeader(RestUtils.Headers.LAST_MODIFIED,
+                  new Date(blobInfo.getBlobProperties().getCreationTimeInMs()));
               Long ifModifiedSinceMs = getIfModifiedSinceMs(restRequest);
               if (ifModifiedSinceMs != null
                   && RestUtils.toSecondsPrecisionInMs(blobInfo.getBlobProperties().getCreationTimeInMs())
                   <= ifModifiedSinceMs) {
                 responseChannel.setStatus(ResponseStatus.NotModified);
-                responseChannel.setHeader(RestUtils.Headers.CONTENT_LENGTH, 0);
```

Comment:
why was content length being set to zero before and why is this being removed now?